disco/extensions/upgrade_simulation/upgrades/common_functions.py

```diff
@@ -327,7 +552,48 @@ def get_all_transformer_info(compute_loading=True, upper_limit=1.5):
     return all_df.reset_index()
 
 
-def get_all_line_info(compute_loading=True, upper_limit=1.5, ignore_switch=True):
+def add_info_line_definition_type(all_df):
+    all_df["line_definition_type"] = "line_definition"
+    all_df.loc[all_df["linecode"] != "", "line_definition_type"] = "linecode"
+    all_df.loc[all_df["geometry"] != "", "line_definition_type"] = "geometry"
+    return all_df
+
+
+def determine_line_placement(line_series):
+    """ Distinguish between overhead and underground cables
+        currently there is no way to distinguish directy using opendssdirect/pydss etc.
+        It is done here using property 'height' parameter and if string present in name
+
+    Parameters
+    ----------
+    line_series
+
+    Returns
+    -------
+    dict
+    """
+    info_dict = {}
+    info_dict["line_placement"] = None
+    if line_series["line_definition_type"] == "geometry":
+            dss.Circuit.SetActiveClass("linegeometry")
+            dss.ActiveClass.Name(line_series["geometry"])
+            h = float(dss.Properties.Value("h"))
+            info_dict["h"] = 0
+            if h >= 0:
+                info_dict["line_placement"] = "overhead"
+            elif h < 0:

```

Comment:
`else:`

Reply:
Yeah, there's no other condition. So changed elif to else.